### PR TITLE
Release stable-2.7.1

### PR DESCRIPTION
**Pull request by alpeb**
This was branched off of `stable-2.7.0`, with the additional changes:

Cherry-picked from master:
- Relax clock skew check (#4195). This is the only user-facing change besides the proxy update.
- Update  helm overrides to match stable ones (#4025) (this only affects integration tests)
- Changes to `go.mod` just for go to not fail in CI:
  - Use correct go module file syntax (#4021)
  - Bump proxy-init to v1.3.2 (#4170)
  - Update linkerd/stern to fix go.mod parsing (#4173)

Other changes:
- Bumped proxy to `v2.91.0`
- In `.github/workflows/workflow.yml` set branch target to `release/stable-2.7` so that CI triggers
- CHANGES.md
- Bumped stable version in values.yaml files (still on the old CI here)

Release notes:

## stable-2.7.1

This release introduces substantial proxy improvements, resulting from continued
profiling & performance analysis. Also support for Kubernetes 1.17 was improved.

To install this release, run: curl https://run.linkerd.io/install | sh

**Full release notes**:

* CLI
  * Relaxed the clock skew check to match the default node heartbeat interval on
    Kubernetes 1.17 and made this check a warning
* Proxy
  * Fixed a bug that could cause log levels to be processed incorrectly
  * Fixed a bug that could cause the proxy's load balancer to stop processing
    updates from service discovery
  * Increased the inbound router's default capacity from 100 to 10k to
    accommodate environments that have a high cardinality of virtual hosts
    served by a single pod
  * Added `request_errors_total` metric exposing the number of requests that
    receive synthesized responses due to proxy errors

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,25 @@
+## stable-2.7.1
+
+This release introduces substantial proxy improvements, resulting from continued
+profiling & performance analysis. Also support for Kubernetes 1.17 was improved.
+
+To install this release, run: curl https://run.linkerd.io/install | sh
```
Comment (kleimkuhler) [inline CHANGES.md:6]: Nit:

```suggestion
To install this release, run: `curl https://run.linkerd.io/install | sh`
```

**Comment by hemakl** [comment]
LGTM!

**Comment by olix0r** [comment]
We're still triaging some issues found during manual pre-release testing. I'll report back here when I have more info.

**Comment by olix0r** [comment]
Are the CI checks stuck on this?

**Comment by alpeb** [comment]
> Are the CI checks stuck on this?

Yeah because this is tied to the old workflow and so clashes with the new required checks set in the github project. The KinD tests went all green though. The cloud check didn't run either, so I'll trigger that manually and I'll report back. It also requires creating the release in Github and pushing the helm release manually. I can assist @kleimkuhler with that.

**Comment by kleimkuhler** [comment]
The two required jobs `Go dependencies` and `Go format` are equivalent to `Validate go deps` and `Go Format`, so those checks are okay.

> The cloud check didn't run either, so I'll trigger that manually and I'll report back

@alpeb I'm not sure if you already kicked something off, but I have integration tests running on a GKE cluster right now.

This PR will need to be admin merged when everything is ready to go!

**Comment by alpeb** [comment]
> @alpeb I'm not sure if you already kicked something off, but I have integration tests running on a GKE cluster right now.

I'm doing the same, we can then compare notes :-)

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+* CLI
+  * Relaxed the clock skew check to match the default node heartbeat interval on
+    Kubernetes 1.17 and made this check a warning
+* Proxy
+  * Fixed a bug that could cause log levels to be processed incorrectly
+  * Fixed a bug that could cause the proxy's load balancer to stop processing
+    updates from service discovery
+  * Increased the inbound router's default capacity from 100 to 10k to
+    accommodate environments that have a high cardinality of virtual hosts
+    served by a single pod
```
Comment (olix0r) [inline CHANGES.md:19]: Actually, this limit has been removed completely now!

Comment (kleimkuhler) [reply CHANGES.md]: Removed!

**Comment by kleimkuhler** [comment]
I had to cherry pick ad4cef48c767dba5bef43f7ac4049bb372472d34 as well since the upgrade certs expired.

**Pull request merged**